python/nav/mibs/itw_mib.py

```diff
@@ -32,947 +31,435 @@
-                                                                reduce_index)
-            self._logger.debug('ItWatchDogsMib:: get_all_sensors: %s = %s',
-                               table, sensors)
-            handler = for_table.map.get(table, None)
-            if not handler:
-                self._logger.error("There is not data handler for %s", table)
-            else:
-                method = getattr(self, handler)
-                result.extend(method(sensors))
+        for table, sensor_groups in self.TABLES.items():
+            self._logger.debug('get_all_sensors: table = %s', table)
+            sensors = yield self.retrieve_table(
+                                        table).addCallback(reduce_index)
+            self._logger.debug('get_all_sensors: %s = %s', table, sensors)
+            for sensor_group in sensor_groups:
+                result.extend(self._handle_sensor_group(sensor_group, sensors))
 
         defer.returnValue(result)
 
-    @classmethod
-    def translate_counter_to_table(cls, counter_name):
-        """Translates the name of a count object under sensorCounts into its
-        corresponding sensor table object name.
-
-        If unable to translate the counter name into a table name, None is
-        returned.
-
-        """
-        counter_bases = [counter_name.replace('Count', '').lower(),
-                         counter_name.replace('SensorCount', '').lower()]
-        suffixes = ['table', 'sensortable', 'monitortable']
-        alternatives = [base + suffix
-                        for base in counter_bases for suffix in suffixes]
 
-        for alt in alternatives:
-            if alt in cls.lowercase_nodes:
-                return cls.lowercase_nodes[alt]
+UNITS = {
+    '%': {'u_o_m': Sensor.UNIT_PERCENT},
+    '0.1 Amps': {
+        'u_o_m': Sensor.UNIT_AMPERES,
+        'precision': 1,
+    },
+    '0.1 Amps (rms)': {
+        'u_o_m': Sensor.UNIT_AMPERES,
+        'precision': 1,
+    },
+    '0.1 Degrees': {
+        'u_o_m': Sensor.UNIT_CELSIUS,
+        'precision': 1,
+    },
+    'Degrees Celsius': {
+        'u_o_m': Sensor.UNIT_CELSIUS,
+    },
+    'kWh': {
+        'u_o_m': Sensor.UNIT_WATTHOURS,
+        'scale': Sensor.SCALE_KILO,
+    },
+    'millivolts': {
+        'u_o_m': Sensor.UNIT_VOLTS_DC,
+        'scale': Sensor.SCALE_MILLI,
+    },
+    'Volt-Amps': {
+        'u_o_m': Sensor.UNIT_VOLTAMPERES,
+    },
+    'Volts': {
+        'u_o_m': Sensor.UNIT_VOLTS_DC,
+    },
+    'Volts (rms)': {
+        'u_o_m': Sensor.UNIT_VOLTS_AC,
+    },
+    'Watts': {
+        'u_o_m': Sensor.UNIT_WATTS,
+    },
+}
+
+
+def getRange(mib, node):
+    res = {}
+    range_ = mib['nodes'][node].get('syntax', {}).get('type', {}).get('range')
+    if not range_:
+        return res
+    if 'min' not in range_ or 'max' not in range_:
+        return res
+    res['minimum'] = range_['min']
+    res['maximum'] = range_['max']
+    return res
+
+
+def convertUnits(mib, node):
```

Comment:
PyLint should already have told you: Please make this function name PEP-8 compliant :-)